lib/SimpleHttpClient/SslClientConnection.cpp

```diff
@@ -302,6 +305,10 @@ bool SslClientConnection::connectSocket() {
 
   _errorDetails.clear();
   _socket = _endpoint->connect(_connectTimeout, _requestTimeout);
+  long flags = fcntl(_socket.fileDescriptor, F_GETFL, 0);
+  if (_isTelemetrics) {
+    fcntl(_socket.fileDescriptor, F_SETFL, flags | O_NONBLOCK);
```

Comment:
Do we want to handle return values from `fcntl` here? It can return an error, at least.